### ISSUE TITLE
Add tests for parent directory files input

### ISSUE THREAD

**Issue by regseb** [stylelint/stylelint#3273]
Hello,

Since stylelint 9.2.0, path with ".." (parent directory) are ignored ; like this example :

```javascript
const config = {
    "rules": { "color-hex-case": "upper" }
};
stylelint.lint({ "files": "../style.css", config }).then(...);
```

To reproduce the problem, download attach file [testcase.tar.gz](https://github.com/stylelint/stylelint/files/1929712/testcase.tar.gz) and execute command :

```shell
tar zxvf testcase.tar.gz
cd testcase/folder
npm install
./testcase.js
```
Output on stylelint 9.2.0 :
> ../style.css { errored: false, output: '[]', results: [] }
> style.css { errored: true,
>   output: '[{"source":"/home/regseb/dev/testcase/folder/style.css","deprecations":[],"invalidOptionWarnings":[],"parseErrors":[],"errored":true,"warnings":[{"line":2,"column":12,"rule":"color-hex-case","severity":"error","text":"Expected \\"#ffffff\\" to be \\"#FFFFFF\\" (color-hex-case)"}]}]',
>   results: 
>    [ { source: '/home/regseb/dev/testcase/folder/style.css',
>        deprecations: [],
>        invalidOptionWarnings: [],
>        parseErrors: [],
>        errored: true,
>        warnings: [Array],
>        ignored: undefined,
>        _postcssResult: [Result] } ] }

Output on stylelint 9.1.3 :

> 
> style.css { errored: true,
>   output: '[{"source":"/home/regseb/dev/testcase/folder/style.css","deprecations":[],"invalidOptionWarnings":[],"parseErrors":[],"errored":true,"warnings":[{"line":2,"column":12,"rule":"color-hex-case","severity":"error","text":"Expected \\"#ffffff\\" to be \\"#FFFFFF\\" (color-hex-case)"}]}]',
>   results: 
>    [ { source: '/home/regseb/dev/testcase/folder/style.css',
>        deprecations: [],
>        invalidOptionWarnings: [],
>        parseErrors: [],
>        errored: true,
>        warnings: [Array],
>        ignored: undefined,
>        _postcssResult: [Result] } ] }
> ../style.css { errored: true,
>   output: '[{"source":"/home/regseb/dev/testcase/style.css","deprecations":[],"invalidOptionWarnings":[],"parseErrors":[],"errored":true,"warnings":[{"line":2,"column":12,"rule":"color-hex-case","severity":"error","text":"Expected \\"#ffffff\\" to be \\"#FFFFFF\\" (color-hex-case)"}]}]',
>   results: 
>    [ { source: '/home/regseb/dev/testcase/style.css',
>        deprecations: [],
>        invalidOptionWarnings: [],
>        parseErrors: [],
>        errored: true,
>        warnings: [Array],
>        ignored: undefined,
>        _postcssResult: [Result] } ] }

**Comment by ntwb** [comment]
After some sleuthing with `git bisect` the cause is commit https://github.com/stylelint/stylelint/commit/7208a87c3c981a93a5a43c31eccca3da00d2ffeb, your testcase works perfectly up unitl that commit @regseb 

That commit turns out to be an update to [globby](https://www.npmjs.com/package/globby) to version 8.0.0 in #3167, now I just need to deermine if it was caused by stylelint or a globby regression.

p.s. Thanks a bunch for the testcase @regseb, extremely helpful 😄

**Comment by ntwb** [comment]
That was quick, indeed it's a globby issue https://github.com/sindresorhus/globby/issues/80, they are waiting for an upstream fix before they can release, I've subscribed to the upstream issues and as soon as its resolved I'll test again, due to semantic versioning fingers crossed an `npm update` will resolve the issue once resolved upstream.

We will also release a _patch_ version of stylelint anyways just to be sure, this might affect more stylelint users though are unaware as the issue is not causing "more" errors and warnings, it's causing less so they might not be being detected at all.

**Comment by mrmlnc** [comment]
Must be fixed `fast-glob@2.2.1`. Please try to install dependencies with `--force` argument.

**Comment by ntwb** [comment]
@mrmlnc Thanks for updating fast-glob` and thanks for the update here :)

@regseb Could you test this out please and let me know how it goes, we can create a new stylelint release if need be but I think `npm update` should cover this, if not try the suggestion above from @mrmlnc to force the update

**Comment by regseb** [comment]
I test with `npm update` : _stylelint 9.2.0_, _globby 8.0.1_ and _fast-glob 2.2.1_. The test case and the real case work.

Thanks.
